File: apps/backend/src/adapters/inbound/http/dto/fuel_log.rs
use chrono::NaiveDate;
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateFuelLogRequest {
    pub log_date: NaiveDate,
    pub odometer: u32,
    pub liters: Decimal,
    pub price_per_liter: Decimal,
    pub station: Option<String>,
    pub is_full_tank: Option<bool>,
}

#[derive(Debug, Serialize)]
pub struct FuelLogResponse {
    pub id: String,
    pub vehicle_id: String,
    pub log_date: NaiveDate,
    pub odometer: u32,
    pub liters: String,
    pub price_per_liter: String,
    /// Serialised as a string to avoid floating-point precision loss.
    pub total_cost: String,
    pub station: Option<String>,
    pub is_full_tank: bool,
}

#[derive(Debug, Serialize)]
pub struct FuelLogListResponse {
    pub logs: Vec<FuelLogResponse>,
}

File: apps/mobile/lib/features/fuel_log/domain/entities/fuel_log.dart
import 'package:decimal/decimal.dart';

/// Domain entity — pure Dart. The data layer maps a DTO into this.
class FuelLog {
  const FuelLog({
    required this.id,
    required this.vehicleId,
    required this.logDate,
    required this.odometer,
    required this.liters,
    required this.pricePerLiter,
    required this.totalCost,
    required this.isFullTank,
    this.station,
  });

  final String id;
  final String vehicleId;
  final DateTime logDate;
  final int odometer;
  final Decimal liters;
  final Decimal pricePerLiter;
  final Decimal totalCost;
  final String? station;
  final bool isFullTank;
}
